### PR TITLE
Remove FirebaseStorageInternal

### PR DESCRIPTION
**Pull request by paulb777**
Now that FirebaseStorage is completely written in Swift, remove all remnants of the Objective C FirebaseStorageInternal.

Since FirebaseStorage now requires GTMSessionFetcher 2.1, it's now incompatible with GoogleSignIn whose GTMAppAuth's dependency is locked to GTMSessionFetcher 1.x (https://github.com/google/GTMAppAuth/issues/174)

I worked around this by pushing GTMSessionFetcher 2.x compatible versions of GoogleSignIn and GTMAppAuth to SpecsDev.

The GoogleMLKit integration test is disabled until it updates to Firebase 10 and GTMSessionFetcher 2.x

### CODE REVIEW

File: SymbolCollisionTest/Podfile [hunk 1]
```diff
@@ -60,12 +61,13 @@ target 'SymbolCollisionTest' do
     pod 'GooglePlacePicker'
     pod 'GooglePlaces'
     pod 'GooglePlusUtilities'
+    #TODO: Uncomment next two when GTMAppAuth updates to GTMSessionFetcher 2.x
```
Comment (ncooke3) [inline SymbolCollisionTest/Podfile:64]: again here if you think it'd be helpful.

File: SymbolCollisionTest/Podfile [hunk 2]
```diff
@@ -28,13 +28,14 @@ target 'SymbolCollisionTest' do
...
     pod 'FirebaseStorage', :path => '../'
-    pod 'FirebaseStorageInternal', :path => '../'
     pod 'GoogleUtilitiesComponents', :path => '../'
 
 #    pod 'FirebaseUI'. - requires use_frameworks!
 
     # Google Pods depending on a Firebase or GDT major version
-    pod 'GoogleMLKit'
+
+    # Uncomment when MLKit updates to Firebase 10 and GTMSessionFetcher 2.x
```
[the remark this answers is not in the document]
Comment (ncooke3) [reply SymbolCollisionTest/Podfile:37]: This idea would then let us grep for Firebase 10 related todos.

**Review by ncooke3 — APPROVED** [review]
Awesome to see!

File: SymbolCollisionTest/Podfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (paulb777) [reply SymbolCollisionTest/Podfile]: Good idea. Done.

File: SymbolCollisionTest/Podfile (hunk 1 above)
Comment (paulb777) [reply SymbolCollisionTest/Podfile]: Actually already resolved. Deleted comment.

**Pull request merged**